### PR TITLE
Fix lint error and handle error

### PR DESCRIPTION
**Pull request by jannyHou**
- [x] Fix eslint failings.
- [x] Comment test cases to get CI pass until we have valid cloudant config.

### CODE REVIEW

File: test/regexp.test.js [hunk 1]
```diff
@@ -27,28 +28,31 @@ describe('cloudant regexp', function() {
...
       entries.should.have.lengthOf(1);
       entries[0].bar.should.equal('b');
       done();
     });
   });
   it('find all foos like b', function(done) {
     Foo.find({where: {bar: {like: 'b'}}}, function(err, entries) {
-      console.log (entries);
+      if (err) return done(err);
+      console.log(entries);
```
Comment (superkhau) [inline test/regexp.test.js]: Why did you move this down 2 lines?

File: test/regexp.test.js [hunk 2]
```diff
@@ -27,28 +28,31 @@ describe('cloudant regexp', function() {
...
       entries[0].bar.should.equal('b');
       done();
     });
   });
   it('find all foos not like b', function(done) {
     Foo.find({where: {bar: {nlike: 'b'}}}, function(err, entries) {
-      console.log (entries);
-      entries.should.have.lengthOf(N-1);
+      if (err) done(err);
+      console.log(entries);
```
Comment (superkhau) [inline test/regexp.test.js]: ditto

File: test/maxrows.test.js [hunk 3]
```diff
@@ -4,25 +4,27 @@
 // License text available at https://opensource.org/licenses/Artistic-2.0
 
 var should = require('should');
+var db, Thing;
```
Comment (superkhau) [inline test/maxrows.test.js]: These weren't there before?

**Comment by superkhau** [comment]
@jannyHou I'm suggesting you break this out into two commits. One for the ESLint specific fixes and one for the error fixes. I'm confused as to which lines of code are for what.

File: test/regexp.test.js (hunk 1 above)
Comment (jannyHou) [inline test/regexp.test.js]: Eslint doesn't only complain syntax error but also errors like "error are supposed to be handled", this is a fix for error not handled. I am not moving this down 2 lines, I am adding a new line `if (err) return done(err)`

File: test/regexp.test.js (hunk 2 above)
Comment (jannyHou) [inline test/regexp.test.js]: forgot a `return` here, will add it then.

File: test/maxrows.test.js (hunk 3 above)
Comment (jannyHou) [inline test/maxrows.test.js]: Nope, this is also reported by eslint, which means you could not use a variable without declaring it.

**Comment by jannyHou** [comment]
@superkhau ok I will break them into different ones. But actually they are all errors reported by eslint, not fixing test cases.

And btw this pr is still in progress, mocha is not recognized in test files when running eslint, I am investigating how to get it work.

**Comment by superkhau** [comment]
Sounds good, besides the missing return, LGTM.

File: test/regexp.test.js (hunk 2 above)
Comment (superkhau) [inline test/regexp.test.js]: BTW, we should actually remove the console log line altogether. But that is outside the scope of this PR.

File: test/regexp.test.js (hunk 1 above)
Comment (superkhau) [inline test/regexp.test.js]: Same as this console log.

File: test/regexp.test.js [hunk 4]
```diff
@@ -27,28 +28,31 @@ describe('cloudant regexp', function() {
   });
   it('find all foos begining with b', function(done) {
     Foo.find({where: {bar: {regexp: '^b'}}}, function(err, entries) {
-      console.log (entries);
+      if (err) return done(err);
+      console.log(entries);
```
Comment (superkhau) [inline test/regexp.test.js]: Ditto for removing.

**Comment by jannyHou** [comment]
@superkhau I just realized that this pr overlaps with the one from @gunjpan : 
https://github.com/strongloop/loopback-connector-cloudant/pull/27

Since he will be back after Sep.19 I am adding fix or commit upon his pr, closing this one now.

**Comment by superkhau** [comment]
I think you should just fix the items I commented on already in this PR.

**Comment by superkhau** [comment]
We can merge yours and he can rebase his later. Sept 11 is much later and these changes are 5 mins and I already reviewed it.

**Comment by jannyHou** [comment]
@superkhau I commented the test cases to get CI pass. PTAL, thanks :)

**Comment by superkhau** [comment]
LGTM, merge when all greenz. ;)

**Comment by superkhau** [comment]
@jannyHou I think this needs backporting -- https://github.com/strongloop/loopback-connector-cloudant/issues/41
